src/components/NotFound/NotFound.js

```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from 'react'
+import { NavLink, useHistory } from 'react-router-dom'
+
+export default function NotFound() {
+  let history = useHistory()
+
+  useEffect(() => {
+    setTimeout(() => {
+      history.push('/')
+    }, 3000)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Return a function from useEffect, that cancels the timeout in case the component unmounts before the redirect.

Reply:
Wow, very attentive. You are absolutely right, added clean up:
```
useEffect(() => {
    const timer = setTimeout(() => {
      history.push('/')
    }, 3000)
    return () => clearTimeout(timer)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])
```